libraries/common/bytearray.effekt

```diff
@@ -106,12 +112,29 @@ extern js """
   }
 """
 
+extern chez """
+  (define (bytearray$compare b1 b2)
+    (let ([len1 (bytevector-length b1)]
+          [len2 (bytevector-length b2)]
+          [minlen (min (bytevector-length b1) (bytevector-length b2))])
+      (let loop ([i 0])
+        (cond
+          [(= i minlen)
+          (cond [(< len1 len2) -1]
+                [(> len1 len2) 1]
+                [else 0])]
+          [(< (bytevector-u8-ref b1 i) (bytevector-u8-ref b2 i)) -1]
+          [(> (bytevector-u8-ref b1 i) (bytevector-u8-ref b2 i)) 1]
+          [else (loop (+ i 1))]))))
+"""
```

Comment:
This is probably further signal that we should really define the comparison in Effekt directly.